apps/src/templates/progress/progressHelpers.js

```diff
@@ -21,6 +29,8 @@ export function lessonIsVisible(lesson, state, viewAs) {
   // Don't show stage if not authorized to see lockable
   if (lesson.lockable && !state.stageLock.lockableAuthorized) {
     return false;
+  } else if (viewAs === ViewType.Teacher) {
```

Comment:
(minor, feel free to ignore) I think this is functionally correct now but doesn't express the intent very clearly.  I think this `if` is more logically attached to lines 36-44 below.  So maybe something like: 

```
if (viewAs === ViewType.Teacher) {
    return true;
} else {
    const hiddentStageState = ...
    ...
    return !isHidden
}
```